### ISSUE TITLE
Typeahead: accessibility issues

### ISSUE THREAD

**Issue by pjhumkhawala** [ng-bootstrap/ng-bootstrap#2871]
### Bug description:
Typeahead component has accessibility issues. 
Code written in the html file:

``` 
<input id="typeahead" type="text" class="form-control" [(ngModel)]="model" [ngbTypeahead]="search" />
```

Code evaluated in the html markup ( from browsers debug console):
``` 
<input _ngcontent-c4="" aria-multiline="false" autocapitalize="off" autocorrect="off" class="form-control ng-untouched ng-pristine ng-valid" id="typeahead" role="combobox" type="text" ng-reflect-ngb-typeahead="function (text$) {="off" aria-autocomplete="list" aria-expanded="false">
```
The HTML markup gives it `role="combobox"` and a property of `aria-multiline="false"`.

However, according to https://www.w3.org/TR/wai-aria-1.1/#combobox, combobox has no arrtibute of type 'aria-multiline' because of which it is failing the axe accessibility test saying that the aria attribute is not allowed.

### Link to minimally-working StackBlitz that reproduces the issue:

https://stackblitz.com/angular/dnbxerkrvpap?file=app%2Ftypeahead-basic.html

(check the markup generated by the browser)

### Versions of Angular, ng-bootstrap and Bootstrap:

Angular CLI: 7.0.6: 

ng-bootstrap: 3.3.1

Bootstrap:4.1.1

**Comment by benouat** [comment]
My 2 cents here, it is true that we are a bit mixing things up. Which is probably disturbing a bit screen readers as a sad side effect 😢 

Ideally as stated by the specification, HTML structure should be looking like

```html
<div role="combobox" aria-expanded="true"  aria-owns="owned_listbox">
    <input type="text" aria-autocomplete="list" aria-controls="owned_listbox"
       aria-activedescendant="selected_option">
</div>
<ul role="listbox" id="owned_listbox">
    <li role="option">Zebra</li>
    <li role="option" id="selected_option">Zoom</li>
</ul>
```

We are clearly missing one extra level of wrapper here, which one should have the `role="combobox"` instead of the `<input >`

**Comment by maxokorokov** [comment]
I don't see how we could fix this in the current implementation of the typeahead, because we simply don't have the wrapper... I'd say we:
- remove the attributes that shouldn't be on the `<input >`
- update all our demos to add the wrapper with those attributres. Something along this:
```ts
```

**Comment by pjhumkhawala** [comment]
@maxokorokov  Will it be possible to let me know when this will be addressed? Also, until this is fixed, can you tell me if there is any workaround to this?

**Comment by maxokorokov** [comment]
@pjhumkhawala unfortunately, I don't think we'll be able to commit to any dates at the moment, but it should be addressed in one of the upcoming minor or major releases. There is clearly a difference between the WAI-ARIA recommendation and what we have.

Could you please tell any particular issues you're having because of this? 
Screen readers don't work? Validators unhappy?

**Comment by aleMaqueda** [comment]
Try removing the attribute :
```
<input type="text" class="form-control" #mainSearch
  [(ngModel)]="currentSearch"  
  [ngbTypeahead]="autocomplete" placeholder="" 
  [resultFormatter]="resultFormatter"
  [inputFormatter]="inputFormatter"
  (selectItem)="selectSuggestion($event)">
```
   ```
@ViewChild('mainSearch') searchInputText: ElementRef;
    constructor(private renderer: Renderer2) {
    ngAfterViewInit(): void {
        if (this.searchInputText !== undefined ) {
            this.renderer.removeAttribute(this.searchInputText.nativeElement, 'aria-multiline');
        }
    }
```

**Comment by gpolychronisAmadeus** [comment]
@benouat @maxokorokov 
Please review the PR.

**Comment by DutchKevv** [comment]
Is this gonna be fixed? It doesn't seem like a huge change

**Comment by BPHvZ** [comment]
Any updates on this?

**Comment by jasonchal** [comment]
Is this issue going to be addressed in an upcoming release?

**Comment by discoinfiltrator** [comment]
https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/combobox_role

> When a combobox's popup is displayed, ensure the [aria-controls](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-controls) attribute on the combobox element is set to the [id](https://developer.mozilla.org/en-US/docs/Web/HTML/Global_attributes#id) of the popup listbox, tree, grid, or dialog element. This is how the relationship between the element with the combobox role and the popup it controls is indicated. (Note: In older ARIA specs, this was aria-owns rather than aria-controls, so you may see aria-owns in older combobox implementations. The aria-owns in the code should be updated to aria-controls!)
